src/backend/cdb/cdbpath.c

```diff
@@ -365,6 +377,13 @@ cdbpath_create_motion_path(PlannerInfo *root,
 	 * materialize nodes on top of motion nodes
 	 */
 
+	/*
+	 * TODO: Check if subpath require parameters(like in PR #12238) when the
+	 * following FIXME from regression tests will be resolved. FIXME: A
+	 * process terminates during execution, see
+	 * https://github.com/greenplum-db/gpdb/issues/10791
+	 */
```

Comment:
@InnerLife0 can you explain this point to me? You mean that if we get to this position in the code in 6X, he backend crashes because of issue#10791? And we should not check parameters here because of it? Or I didn't get the idea?

Reply:
Placing the `if` condition here as it was done in master, cause a [crash](https://github.com/greenplum-db/gpdb/pull/12238#issuecomment-871810154) of one regression test, which points to issue#10791. This bug is known( unlike master, the next regression test is commented out), but fixing it is not a goal of this issue. So we point we need to do the check, but only after another issue will be fixed.

Reply:
I am still confused with this explanation. An issue [10791](https://github.com/greenplum-db/gpdb/issues/10791) is not about failing on a building plan stage, but rather about incorrect plan execution. You fail on a build plan phase, but it can be fixed with a patch like
```
diff --git a/src/backend/cdb/cdbpath.c b/src/backend/cdb/cdbpath.c
index 3a6e8138303..4905cdbf479 100644
--- a/src/backend/cdb/cdbpath.c
+++ b/src/backend/cdb/cdbpath.c
@@ -378,11 +378,10 @@ cdbpath_create_motion_path(PlannerInfo *root,
         */
 
        /*
-        * TODO: Check if subpath require parameters(like in PR #12238) when the
-        * following FIXME from regression tests will be resolved. FIXME: A
-        * process terminates during execution, see
-        * https://github.com/greenplum-db/gpdb/issues/10791
+        * If the subpath requires parameters, we cannot generate Motion atop of it.
         */
+       if (!bms_is_empty(PATH_REQ_OUTER(subpath)))
+               return NULL;
 
        /* Create CdbMotionPath node. */
        pathnode = makeNode(CdbMotionPath);
diff --git a/src/backend/optimizer/util/pathnode.c b/src/backend/optimizer/util/pathnode.c
index 969292b07f5..496f8f7a04c 100644
--- a/src/backend/optimizer/util/pathnode.c
+++ b/src/backend/optimizer/util/pathnode.c
@@ -1862,6 +1862,8 @@ create_unique_path(PlannerInfo *root, RelOptInfo *rel, Path *subpath,
 
                locus = cdbpathlocus_from_exprs(root, uniq_exprs, opfamilies, numsegments);
         subpath = cdbpath_create_motion_path(root, subpath, NIL, false, locus);
+               if (!subpath)
+                       return NULL;
                /*
                 * We probably add agg/sort node above the added motion node, but it is
                 * possible to add an agg/sort node below this motion node also,
```
I still think that we should not remove this default case check but rather fix this case.

Reply:
We've already discussed this problem with team on one of the voice chat sessions. This is not the only place were NULL value returned by `cdbpath_create_motion_path` is ignored. We decided this fix should not be a part of a current PR, but should be [mentioned](https://github.com/greenplum-db/gpdb/pull/12287#issuecomment-877009132) publicly. The only better way we could do is to create a separate PR by ourselves.

Reply:
@InnerLife0 can you write the arguments to leave this PR partially fixed and do the rest of the job in a separate PR instead of doing a full fix? I suppose I have missed that voice chat.

Reply:
Left a comment in PR head.